caiman/source_extraction/volpy/utils.py

```diff
@@ -0,0 +1,281 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Mar 23 16:45:00 2020
+This file create functions used for demo_pipeline_voltage_imaging
+@author: caichangjia
+"""
+#%% 
+import os
+from IPython import get_ipython
+import matplotlib.pyplot as plt
+from matplotlib.widgets import Slider
+import numpy as np
+import tensorflow as tf
+import caiman as cm
+from caiman.external.cell_magic_wand import cell_magic_wand_single_point
+from caiman.paths import caiman_datadir
+
+def correlation_image(fnames, fr):
+    """ Compute correlation image with Gaussian Blur. The method is relative slow
+    in speed, but it is suitable for high noise movie.
+    Args:
+        fnames: 3-D array
+            motion corrected movie in F-order memory mapping format
+        
+        fr: int
+            frame rate
+            
+    Return: 
+        Cn: 2-D array
+            correlation image
+    """
+    m = cm.load(fnames, fr=fr)
+    ma = m.computeDFF(secsWindow=1)[0]
+    Cn = ma.copy().gaussian_blur_2D().local_correlations(swap_dim=False, 
+                eight_neighbours=True, frames_per_chunk=1000000000)
+    return Cn
+
+def quick_annotation(img, min_radius, max_radius, roughness=2):
+    """ Quick annotation method in VolPy using cell magic wand plugin
+    Args:
+        img: 2-D array
+            img as the background for selection
+            
+        min_radius: float
+            minimum radius of the selection
+            
+        max_radius: float
+            maximum raidus of the selection
+            
+        roughness: int
+            roughness of the selection surface
+            
+    Return:
+        ROIs: 3-D array
+            region of interests 
+            (# of components * # of pixels in x dim * # of pixels in y dim)
+    """
+    get_ipython().run_line_magic('matplotlib', 'auto')
```

Comment:
would this work with regular python?

Reply:
What I know is it works on Spyder. It serves the function to jump out a window (matplotlib auto) for user manually selecting neurons. Is there a better choice?

Reply:
Spyder uses ipython as its background. You might want to test this code from the regular python commandline. I think one of the existing demos demonstrates how to do this with regular matplotlib code.

Reply:
(I think this is the last thing that needs some attention in this diff; otherwise looking good)

Reply:
looking good now